draco/asp/helpers.lp

```diff
@@ -60,10 +67,25 @@ attribute(mark_channel_discrete_or_binned,M,C) :-
     attribute((encoding,channel),E,C),
     attribute((encoding,binning),E,_).
 
+% @helper(mark_encoding_discrete_or_binned) Whether a mark has a discrete scale or binned encoding for an encoding.
+attribute(mark_encoding_discrete_or_binned,M,E) :-
+    entity(encoding,M,E),
+    attribute((encoding,channel),E,C),
+    attribute(mark_scale_channel,M,T,C),
+    domain(discrete_scale,T).
+attribute(mark_encoding_discrete_or_binned,M,E) :-
+    entity(encoding,M,E),
+    attribute((encoding,binning),E,_).
+
 % @helper(encoding_cardinality,E,N) The cardinality of an encoding.
 attribute(encoding_cardinality,E,N) :-
     attribute((field,unique),F,N),
     attribute((encoding,field),E,F),
     not attribute((encoding,binning),E,_).
 attribute(encoding_cardinality,E,N) :-
     attribute((encoding,binning),E,N).
+
+% @helper(mark_has_aggregate) Whether a mark has aggregate encoding.
+attribute(mark_has_aggregate,M) :-
```

Comment:
I don't think we need this helper as we can just check `attribute((encoding,aggregate),E,_).`. Let's remove it. In general, we should only add helpers if it makes a big difference as every helper makes it harder to read the code.